templates/common/_base/files/ofport-request.yaml

```diff
@@ -0,0 +1,114 @@
+mode: 0755
+path: "/etc/NetworkManager/dispatcher.d/pre-up.d/10-ofport-request.sh"
+contents:
+  inline: |
+    #!/bin/bash
+    # Set interface ofport_request to guarantee stable ofport numbers. This is important for flow matches.
+    # Otherwise, another ofport number is assigned to the interface on every restart of NetworkManager.
+    # This script will build a map INTERFACE_NAME->ofport_request and will save it to file CONFIGURATION_FILE.
+    # When an interface is brought up, this will reuse the value from the map if such a value exists.
+    # Otherwise, this will try to use the current ofport value. If the ofport value is already reserved, then
+    # this use the lowest available numerical value, instead.
+    set -eux -o pipefail
+    if [[ "{{ .NetworkType }}" != "OVNKubernetes" ]]; then
+        exit 0
+    fi
+    
+    INTERFACE_NAME=$1
+    OPERATION=$2
+
+    # Only execute this on pre-up
+    if [ "${OPERATION}" != "pre-up" ]; then
+        exit 0
+    fi
+    
+    # Get the interface's NM uuid
+    INTERFACE_CONNECTION_UUID=$(nmcli -t -f  device,type,uuid conn | awk -F ':' '{if($1=="'${INTERFACE_NAME}'" && $2!~/^ovs*/) print $NF}')
+    if [ "$INTERFACE_CONNECTION_UUID" == "" ]; then
+        exit 0
+    fi
+    
+    # Get the interface's slave-type. If this is not an ovs-port, then exit
+    INTERFACE_OVS_SLAVE_TYPE=$(nmcli -t -f connection.slave-type conn show "$INTERFACE_CONNECTION_UUID" | awk -F ':' '{print $NF}')
+    if [ "$INTERFACE_OVS_SLAVE_TYPE" != "ovs-port" ]; then
+        exit 0
+    fi
+    
+    # This is not necessarily a UUID (can be a name in case of bonds) but this should be unique
+    PORT=$(nmcli -t -f connection.master conn show "$INTERFACE_CONNECTION_UUID" | awk -F ':' '{print $NF}')
+    if [ "$PORT" == "" ]; then
+        exit 0
+    fi
+    
+    # Get the interface's NM uuid
+    PORT_CONNECTION_UUID=$(nmcli -t -f  device,type,uuid conn | awk -F ':' '{if( ($1=="'${PORT}'" || $3=="'${PORT}'") && $2~/^ovs*/) print $NF}')
+    if [ "$PORT_CONNECTION_UUID" == "" ]; then
+        exit 0
+    fi
+    
+    # Get the port's slave-type. If this is not an ovs-bridge, then exit
+    PORT_OVS_SLAVE_TYPE=$(nmcli -t -f connection.slave-type conn show "$PORT_CONNECTION_UUID" | awk -F ':' '{print $NF}')
+    if [ "$PORT_OVS_SLAVE_TYPE" != "ovs-bridge" ]; then
+        exit 0
+    fi
+    
+    # Get the port's master = bridge name
+    BRIDGE_NAME=$(nmcli -t -f connection.master conn show "$PORT_CONNECTION_UUID" | awk -F ':' '{print $NF}')
+    # Limit this to br-ex and br-ex1 only. If one wanted to enable this for all OVS bridges,
+    # the condition would be: if [ "$BRIDGE_NAME" == "" ]; then
+    if [ "$BRIDGE_NAME" != "br-ex" ] && [ "$BRIDGE_NAME" != "br-ex1" ]; then
+        exit 0
+    fi
+    
+    CONFIGURATION_FILE=/run/ofport_requests.${BRIDGE_NAME}
+    
+    # Declare a new map. If CONFIGURATION_FILE exists, source entries from there
```

Comment:
this is really an array. Why use an array to store a single value? There can only be one physical port at a time right?

Reply:
a) I wrote the script on a RHEL VM and purposefully added lots of interfaces and interface variatoins to make the script more capable, flexible and future proof. I did not want the script to be a point of failure if for some reason we decide to change anything (as unlikely as it may be).

b) As a more compelling argument, OVS bonds are implemented based on flows and they consist of multiple interfaces each with their own index, so they look like this:
~~~
    Bridge br-ex
        Port bond0
            Interface eth4
                type: system
            Interface eth5
                type: system
~~~

And each of them get their own port ID:
~~~
 5(eth4): addr:52:54:00:df:9c:35
     config:     0
     state:      0
     speed: 0 Mbps now, 0 Mbps max
 6(eth5): addr:52:54:00:15:ec:29
     config:     0
     state:      0
     speed: 0 Mbps now, 0 Mbps max
~~~

So the array solution is a must if we don't want to exclude OVS bonding. Granted, ovs-configure.sh doesn't do ovs-bonding at the moment, but we don't know for how long this script would be around and what changes we make to ovs-configure.sh, so me might as well do it right in this script here and keep it general purpose and have it handle anything we throw at it.